pankti-extract/src/main/java/se/kth/castor/pankti/extract/util/MethodUtil.java

```diff
@@ -25,4 +47,97 @@ else if (paramType.contains("double"))
         else paramSignature.append('L').append(paramType.replace('.', '/')).append(';');
         return paramSignature.toString();
     }
+
+    /**
+     * Checks that the invocation target for a nested method invocation is a field defined in the declaring type of the method.
+     *
+     * @param method     The method with the nested method invocation
+     * @param invocation The nested method invocation
+     * @return true if the invocation target is a final or static field in the declaring type of the method
+     */
+    private static boolean isInvocationTargetAFinalStaticField(final CtMethod<?> method,
+                                                               final CtInvocation<?> invocation) {
+
+        CtField<?> field = method.getDeclaringType().getField(String.valueOf(invocation.getTarget()));
+        // Invocation target is a field and not a local variable
+        if (field != null
+                && invocation.getTarget().getElements(new TypeFilter<>(CtFieldReferenceImpl.class)).size() == 1) {
+            return field.getModifiers().contains(ModifierKind.STATIC)
+                    || field.getModifiers().contains(ModifierKind.FINAL);
+        }
+        return true;
+    }
```

Comment:
I don't think this is implemented according to the method documentation.

If `field` is `null` (indicating that `invocation` does not target a field), this will cause the method to return `true`, as per the final return statement. Thus, this method returns `true` if `invocation` targets a final/static field in `method`, _or_ if the invocation _doesn't_ target a field in `method`. To match the docs, the final return statement should rather return `false`.

Now, this will cause some test failures, but only because there's no other check in `findNestedMethodCalls` that the invocation target is actually a field read.

I see two ways to fix this:

1. Change the final return to `false` and add augment the invocation site in `findNestedMethodCalls` to `thisInvocation.getTarget() instanceof CtFieldRead && !isInvocationTargetAFinalStaticField(method, thisInvocation)`.
2. Change this method to check that the invocation target _is_ a non-final, non-static field read (i.e. `isInvocationTargetNonFinalNonStaticField`), and remove the negation in the call in `findNestedMethodCalls`.

Reply:
Yes, this was bothering me a bit too! Thanks for the fix!